### ISSUE TITLE
XMLWriter should implement 'Closeable' or 'AutoCloseable' interfaces

### ISSUE THREAD

**Issue by ghost** [dom4j/dom4j#120]
Java 7 added the new ARM (automatic resource management) syntax for try-catch statements, in order to support said ARM syntax, XMLWriter should implement either one of the two closeable interfaces, preferably `Closeable`, to avoid having to use the now outdated try-catch-finally syntax for safe closing of the underlying writer.

**Comment by IOsetting** [comment]
Is there a plan to add this?

**Comment by FilipJirsak** [comment]
Yes, using `Closeable` or `AutoCloseable` for `XMLWriter` makes sense. Pullrequests are welcome.

**Comment by reinhapa** [comment]
@FilipJirsak I've created PR #160  to implement this request 😎
